WolvenKit/Views/Shell/DockingAdapter.xaml.cs

```diff
@@ -476,25 +494,43 @@ private void PART_DockingManager_ActiveWindowChanged_1(DependencyObject d, Depen
             _viewModel?.UpdateTitle();
         }
 
-        private void PART_DockingManager_OnCloseAllTabs(object sender, CloseTabEventArgs e)
+        private async void PART_DockingManager_OnCloseAllTabs(object sender, CloseTabEventArgs e)
         {
             var closeTasks = e.ClosingTabItems.OfType<TabItemExt>()
                 .Select(item => item.Content).OfType<ContentPresenter>()
                 .Select(contentPresenter => contentPresenter.Content).OfType<ContentControl>()
                 .Select(contentControl => contentControl.Content).OfType<DocumentViewModel>()
-                .Select(async vm =>
+                .Select(TryCloseDocumentAsync);
+
+            var results = await Task.WhenAll(closeTasks);
+
+            e.Cancel = results.Any(x => !x);
+        }
+
+        public async Task<bool> CloseAllAsync()
+        {
+            if (_viewModel == null)
+            {
+                return true;
+            }
+
+            var allClosed = true;
+
+            for (var i = _viewModel.DockedViews.Count - 1; i >= 0; i--)
```

Comment:
Would it not make sense to do an await Task.WhenAll here, rather than serially awaiting?